### PR TITLE
[clang][driver] Cleanup UEFI toolchain driver

### PR DESCRIPTION
**Pull request by Prabhuk**
Updating UEFI header includes to not include system include directories, adding includes from compiler resource directory and minor cleanups.

### CODE REVIEW

File: clang/lib/Driver/ToolChains/UEFI.cpp [hunk 1]
```diff
@@ -35,6 +35,24 @@ UEFI::UEFI(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
...
 
+void UEFI::AddClangSystemIncludeArgs(const ArgList &DriverArgs,
+                                     ArgStringList &CC1Args) const {
+  if (DriverArgs.hasArg(options::OPT_nostdinc))
+    return;
+
+  if (!DriverArgs.hasArg(options::OPT_nobuiltininc)) {
+    SmallString<128> Dir(getDriver().ResourceDir);
+    llvm::sys::path::append(Dir, "include");
+    addSystemInclude(DriverArgs, CC1Args, Dir.str());
```
Comment (Prabhuk) [inline clang/lib/Driver/ToolChains/UEFI.cpp:46]: @petrhosek -- Introduced this to include headers from clang resource directory. Rest of this method implementation is copied from Baremetal driver without much thought. PTAL.

**Comment by Prabhuk** [comment]
@RossComputerGuy FYI

**Review by petrhosek — APPROVED** [review]

**Comment by Prabhuk** [comment]
@RossComputerGuy -- I am planning to land this today. Would you like to take a look and let me know if you are ok with these clean ups? I am still not able to add you as a reviewer. Maybe something to do with getting an approval? I'll check the process docs.

File: clang/lib/Driver/ToolChains/UEFI.cpp (hunk 1 above)
Comment (jhuber6) [reply clang/lib/Driver/ToolChains/UEFI.cpp:46]: Should be fine so long as it includes `lib/<triple/` and `lib/clang/20/lib/<triple>`. Do we have an existing clang driver UEFI target test?

**Review by RossComputerGuy — APPROVED** [review]
Looks good to me.

File: clang/lib/Driver/ToolChains/UEFI.cpp (hunk 1 above)
Comment (Prabhuk) [reply clang/lib/Driver/ToolChains/UEFI.cpp:46]: We have a minimal test here: https://github.com/llvm/llvm-project/blob/main/clang/test/Driver/uefi-constructed-args.c

This is the header search paths with this PR. My original goal of this change was to remove `/usr/include` and `/usr/local/include` from the search list.

```
[hi on] prabhukr@prabhukr:~/llvm-builds/b2$ ./bin/clang -E -v --target=x86_64-uefi - < /dev/null
Fuchsia clang version 20.0.0git (git@github.com:prabhuk/llvm-project.git e30c180214f2ca102ba6fcf1b542db7c5f2ac5fd)
Target: x86_64-unknown-uefi
Thread model: posix
InstalledDir: /usr/local/google/home/prabhukr/llvm-builds/b2/bin
Build config: +assertions
 (in-process)
 "/usr/local/google/home/prabhukr/llvm-builds/b2/bin/llvm" "clang" -cc1 -triple x86_64-unknown-uefi -E -disable-free -clear-ast-before-backend -main-file-name - -mrelocation-model pic -pic-level 2 -mframe-pointer=all -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -debugger-tuning=gdb -fdebug-compilation-dir=/usr/local/google/home/prabhukr/llvm-builds/b2 -v -fcoverage-compilation-dir=/usr/local/google/home/prabhukr/llvm-builds/b2 -resource-dir /usr/local/google/home/prabhukr/llvm-builds/b2/lib/clang/20 -internal-isystem /usr/local/google/home/prabhukr/llvm-builds/b2/lib/clang/20/include -ferror-limit 19 -fgnuc-version=4.2.1 -fskip-odr-check-in-gmf -fcolor-diagnostics -faddrsig -o - -x c -
clang -cc1 version 20.0.0git based upon LLVM 20.0.0git default target x86_64-unknown-linux-gnu
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/google/home/prabhukr/llvm-builds/b2/lib/clang/20/include
End of search list.
# 1 "<stdin>"
# 1 "<built-in>" 1
# 1 "<built-in>" 3
# 389 "<built-in>" 3
# 1 "<command line>" 1
# 1 "<built-in>" 2
# 1 "<stdin>" 2

```

**Pull request merged**